### ISSUE TITLE
Add support for gofiber

### ISSUE THREAD

**Issue by frederikhors** [99designs/gqlgen#1664]
I think we should add support for framework like [GoFiber](https://github.com/gofiber/fiber).

Someone has already tried: https://github.com/Just4Ease/fiber-gqlgen/blob/master/handler.go.

**Comment by numToStr** [comment]
I originally requested the support and if gofiber gets the official from gqlgen then it will just make gofiber more powerful.

**Comment by frederikhors** [comment]
> However, I note there's a forked version of gqlgen with support at https://github.com/arsmn/fastgql.

Is old, doesn't work and there is no docs.

**Comment by TheBeachMaster** [comment]
Cross-posting this for anyone stuck here's the solution.
[Gofiber](https://github.com/gofiber/) has an [adaptor middleware](https://github.com/gofiber/adaptor) that can convert between net http handlers and Fiber handlers. 

##TL;DR
Use `adaptor.HandleHTTP()` for main GraphQL service and `adaptor.HTTPHandlerFunc()` for playground

```go
/// graphql.go
package mypackage
import (
 "github.com/99designs/gqlgen/graphql/handler"
// Some of your fancy imports
)
func NewGraphQLServer(whatever ...interface{}) *handler.Server {
graphqlServer := handler.NewDefaultServer(...whatever)
// Something else
return graphqlServer
}
```
Then use it like so in your `fiber.Handler` logic
```go
/// routes.go
import (
"github.com/gofiber/adaptor/v2"
)
func GraphQLService(whatever ...interface{}) fiber.Handler {
 srv := mypackage.NewGraphQLServer(stuff ...)
return adaptor.HTTPHandler(srv)
}
```

For playground use `adaptor.HTTPHandlerFunc` instead as shown
```go
/// routes.go
import (
"github.com/gofiber/adaptor/v2"
 "github.com/99designs/gqlgen/graphql/playground"
)
func Playground() fiber.Handler {
pg := playground.Handler("Graphql Playground", "/graphql")
return adaptor.HTTPHandlerFunc(pg)
}
```

**Comment by shinebayar-g** [comment]
This is a final missing piece of fiber x gqlgen.

**Comment by alexander-ushakov-93** [comment]
It will be very nice to have official support of Fiber/fasthttp.

**Comment by gaby** [comment]
Any news on this?

**Comment by shyaaam** [comment]
@gaby  It worked for me using the following code. This boilerplate code was written by ChatGPT (while i just wrote the  _wrapHandler_ func to make it typesafe compatible). This works for me, at least I didn't run into any complications so far. You can try and let me know.

```
package main

import (
	"your-module/graph" // Import the generated code from gqlgen

	"net/http"

	"github.com/gofiber/fiber/v2"
	"github.com/99designs/gqlgen/graphql/handler"
	"github.com/99designs/gqlgen/graphql/playground"
	"github.com/valyala/fasthttp/fasthttpadaptor"
)

func wrapHandler(f func(http.ResponseWriter, *http.Request)) func(ctx *fiber.Ctx) {
	return func(ctx *fiber.Ctx) {
		fasthttpadaptor.NewFastHTTPHandler(http.HandlerFunc(f))(ctx.Context())
	}
}

func main() {
	app := fiber.New()

	// Create a gqlgen handler
	h := handler.NewDefaultServer(graph.NewExecutableSchema(graph.Config{Resolvers: &graph.Resolver{}}))


	// Serve GraphQL API
	app.Post("/graphql", func(c *fiber.Ctx) error {
		wrapHandler(h.ServeHTTP)(c)
		return nil
	})

	// Serve GraphQL Playground
	app.Get("/playground", func(c *fiber.Ctx) error {
		wrapHandler(playground.Handler("GraphQL", "/graphql"))(c)
		return nil
	})

	// Start the server
	app.Listen(":3000")
}

```

**Comment by oSethoum** [comment]
@shyaaam  when using the adapter, does it mean the speed decreases from gofiber to net/http?

**Comment by giancarlosisasi** [comment]
As mentioned in the above comment, I have used the adaptor to integrate fiber with gqlgen. But when following the subscription recibe from the documentation page, I got an error saying that the request can’t be upgraded 😞
Maybe there is a web framework or router that is officially recommended by gqlgen? 😟

**Comment by UnAfraid** [comment]
> As mentioned in the above comment, I have used the adaptor to integrate fiber with gqlgen. But when following the subscription recibe from the documentation page, I got an error saying that the request can’t be upgraded 😞 Maybe there is a web framework or router that is officially recommended by gqlgen? 😟

I've always used chi router that works fine with gqlgen, used it quite a bit in production on very large api with over 400+ types

**Comment by JonasDoe** [comment]
> [@shyaaam](https://github.com/shyaaam) when using the adapter, does it mean the speed decreases from gofiber to net/http?

I'ld expect so - from what I understand, `fiber`/`fasthttp` tries to squeeze out performance by pooling reusable pointers ... and those will get "replaced" right here with old-fashioned `net/http` logic. The [documentation](https://docs.gofiber.io/middleware/adaptor) also warns you about the overhead. Use it for compatibility aspects (or if you dig their API), but expecting performance gains is probably moot. For this reason I gave the `adaptor` a wide berth. If you're commited to `net/http` for whatever reason, just stick with it.

I'm not really in the details of `fiber's`/`fasthttp's` working, but I think as soon as you have an abstraction layer, you'll lose much of the benefits. Have a look at the first section [here](https://github.com/valyala/fasthttp?tab=readme-ov-file#faq) - and see how these "optimzation opportunities" won't make it through the `adaptor`.

The only viable solution would be a dedicated fiber implementation of the gql server - either by adding `fiber` als dependency (and I'm not sure that's a good idea), or by forking the project (which doesn't strike me as a great idea either). :/
